openeo_processes_dask/process_implementations/cubes/general.py

```diff
@@ -32,3 +34,32 @@ def dimension_labels(data: RasterCube, dimension: str) -> RasterCube:
             f"Provided dimension ({dimension}) not found in data.dims: {data.dims}"
         )
     return data.coords[dimension]
+
+
+def add_dimension(
+    data: RasterCube, name: str, label: str, type: Optional[str] = "other"
```

Comment:
`type` is currently ignored. Ideally this would be worked into the `.openeo` accessor in `_xr_interop.py`, as an extra list for each type of dimension (e.g. "known_spatial_dims", etc.) that the name of this dimension is appended to and considered when `.openeo.spatial_dims` is called.

Reply:
Done, see last commit.